src/utils/CMakeLists.txt

```diff
@@ -5,17 +5,30 @@ set(utils_src_files
     csv_parsing.cpp
     file.cpp
     file_locker.cpp
-    license.cpp
     memory.cpp
     memory_tracker.cpp
     readable_size.cpp
-    settings.cpp
     signals.cpp
     sysinfo/memory.cpp
     temporal.cpp
     thread.cpp
     thread_pool.cpp
     uuid.cpp)
 
+find_package(Boost REQUIRED)
+
 add_library(mg-utils STATIC ${utils_src_files})
-target_link_libraries(mg-utils mg-kvstore mg-slk stdc++fs Threads::Threads spdlog fmt gflags uuid rt)
+target_link_libraries(mg-utils PUBLIC Boost::headers fmt::fmt spdlog::spdlog)
+target_link_libraries(mg-utils PRIVATE stdc++fs Threads::Threads gflags uuid rt)
```

Comment:
If it's used to compile only source files inside the `mg-utils` why should we implicitly expose them to other libs?
I mean with static libs shouldn't this only expose includes and flags/defs? For symbols, t should use only the ones it needs and throw everything else anyways.

Reply:
I think no. Lets consider the following case:

```cmake
add_library(A STATIC A.hpp A.cpp)

add_library(B STATIC B.hpp B.cpp)
# A.hpp in only included in B.cpp, not in the header
target_link_libraries(B PRIVATE A)

...

# somewhere else
add_executable(E E.hpp E.cpp)
target_link_libraries(E B)
```

This would fail, because when it tries to link `E`, then the objects in `A` cannot be resolved, as `E` is not linked against `A`. So basically the private dependencies are not propagated and this can cause a problem. Or at least that's how [I interpreted it](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents). In my opinion this is not a good thing.

> I mean with static libs shouldn't this only expose includes and flags/defs?
And also transitively links the dependents to the libs that are necessary for the static lib.

> For symbols, t should use only the ones it needs and throw everything else anyways.
I think the symbols from the dependencies are not included into the dependent static lib.

I will try this out to be 100% sure, because I am in doubt.

Reply:
Based on what I got with Pulsar, cmake link keywords don't affect the symbols or anything on the linking itself, but it defines which libraries can CMake use for building a specific lib.
E.g.
If E used something from A, `PUBLIC` would allow cmake to get A as dependency transitively from B. If it links and uses only symbols from B it should be able to get them, while B as a static lib should have everything it needs from A included in itself. `PRIVATE` won't hide the symbols from A that B incorporated in itself, but it won't have all the symbols, only the ones it required.

Reply:
But definitely check it out!

Reply:
You are right, sorry.